src/qt/proposalwizard.h

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2025 The Dash Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_QT_PROPOSALWIZARD_H
+#define BITCOIN_QT_PROPOSALWIZARD_H
+
+#include <QDialog>
+#include <QCloseEvent>
+#include <QObject>
+#include <QString>
+#include <qt/forms/ui_proposalwizard.h>
+#include <memory>
+
+class QDateTimeEdit;
+class QDoubleSpinBox;
+class QLabel;
+class QLineEdit;
+class QComboBox;
+class QPlainTextEdit;
+class QPushButton;
+class QStackedWidget;
+class QProgressBar;
+class QTimer;
+
+namespace interfaces { class Node; }
+class WalletModel;
+// The UI header is included above for complete type to satisfy unique_ptr deleter
+
+class ProposalWizard : public QDialog
+{
+    Q_OBJECT
+public:
+    explicit ProposalWizard(interfaces::Node& node, WalletModel* walletModel, QWidget* parent = nullptr);
+    ~ProposalWizard();
+
+private Q_SLOTS:
+    void onNextFromDetails();
+    void onBackToDetails();
+    void onValidateJson();
+    void onNextFromReview();
+    void onBackToReview();
+    void onPrepare();
+    void onMaybeAdvanceAfterConfirmations();
+    void onSubmit();
+    void onGoToSubmit();
+
+private:
+    interfaces::Node& m_node;
+    WalletModel* m_walletModel;
+    std::unique_ptr<Ui::ProposalWizard> m_ui;
+
+    // UI pointers
+    QStackedWidget* stacked{nullptr};
+    QLineEdit* editName{nullptr};
+    QLineEdit* editUrl{nullptr};
+    QLineEdit* editPayAddr{nullptr};
+    QLineEdit* editTxid{nullptr};
+    QComboBox* comboFirstPayment{nullptr};
+    QComboBox* comboPayments{nullptr};
+    QDoubleSpinBox* spinAmount{nullptr};
+    QLabel* labelFeeValue{nullptr};
+    QLabel* labelTotalValue{nullptr};
+    QLabel* labelSubheader{nullptr};
+    QLabel* labelPrepare{nullptr};
+    QPushButton* btnNext1{nullptr};
+    QPushButton* btnPrepare{nullptr};
+
+    QPlainTextEdit* plainJson{nullptr};
+    QLineEdit* editHex{nullptr};
+    QLabel* labelValidateStatus{nullptr};
+    QPushButton* btnNext2{nullptr};
+
+    QLabel* labelTxid{nullptr};
+    QLabel* labelConf{nullptr};
+    QLabel* labelConfStatus{nullptr};
+    QLabel* labelEta{nullptr};
+    QProgressBar* progressConfirmations{nullptr};
+    QLabel* labelConfStatus2{nullptr};
+    QLabel* labelEta2{nullptr};
+    QProgressBar* progressConfirmations2{nullptr};
+    QPushButton* btnNext3{nullptr};
+
+    QLabel* labelGovObjId{nullptr};
+    QLineEdit* editGovObjId{nullptr};
+    QPushButton* btnCopyGovId{nullptr};
+    QPushButton* btnSubmit{nullptr};
```

Comment:
there're 30 members with different UI elements and no descriptions - which one is which one and which stage it belongs, not saying about poor naming - "labelConfSatus2", "labelConf", "labelConfStatus" (which one is which one?).

Also, btnNext1, Next2, bntNext3, btnSubmit - is it really should be new buttons everytime or can be a same button that change top of form? I think forms usually have main view that changes; but buttons Back&Next - just static and switch an internal state of form